silkworm/capi/silkworm.h

```diff
@@ -95,6 +95,39 @@ struct SilkwormChainSnapshot {
     struct SilkwormTransactionsSnapshot transactions;
 };
 
+struct SilkwormInvertedIndexSnapshot {
+    struct SilkwormMemoryMappedFile segment;         // .ef
+    struct SilkwormMemoryMappedFile accessor_index;  // .efi
+};
+
+struct SilkwormHistorySnapshot {
+    struct SilkwormMemoryMappedFile segment;         // .v
+    struct SilkwormMemoryMappedFile accessor_index;  // .vi
+    struct SilkwormInvertedIndexSnapshot inverted_index;
+};
+
+struct SilkwormDomainSnapshot {
+    struct SilkwormMemoryMappedFile segment;          // .kv
+    struct SilkwormMemoryMappedFile existence_index;  // .kvei
+    struct SilkwormMemoryMappedFile btree_index;      // .bt
+    bool has_accessor_index;
+    struct SilkwormMemoryMappedFile accessor_index;  // .kvi
+    bool has_history;
+    struct SilkwormHistorySnapshot history;
+};
+
+struct SilkwormStateSnapshot {
+    struct SilkwormDomainSnapshot account;
+    struct SilkwormDomainSnapshot storage;
+    struct SilkwormDomainSnapshot code;
+    struct SilkwormDomainSnapshot commitment;
+    struct SilkwormDomainSnapshot receipt;
+    struct SilkwormInvertedIndexSnapshot log_address;
+    struct SilkwormInvertedIndexSnapshot log_topic;
+    struct SilkwormInvertedIndexSnapshot trace_from;
+    struct SilkwormInvertedIndexSnapshot trace_to;
```

Comment:
```suggestion
    struct SilkwormDomainSnapshot accounts;
    struct SilkwormDomainSnapshot storage;
    struct SilkwormDomainSnapshot code;
    struct SilkwormDomainSnapshot commitment;
    struct SilkwormDomainSnapshot receipts;
    struct SilkwormInvertedIndexSnapshot log_address;
    struct SilkwormInvertedIndexSnapshot log_topics;
    struct SilkwormInvertedIndexSnapshot traces_from;
    struct SilkwormInvertedIndexSnapshot traces_to;
```
for naming consistency with state/schema_config.hpp `kDomainName*` and `kInvIdxName*`.